### PR TITLE
feat: openai streaming spans show up in the ui

### PR DESCRIPTION
**Pull request by axiomofjoy**
- creates a new span only once upon end of iteration
- only creates an event for the first streaming token event, not for each event (otherwise, there are too many events to make sense)
- fix bug preventing span updates, although span updates are not used in this PR
- add notebook for running openai sdk requests of various request types

Trace

Screenshot 2023-12-07 at 11 10 21 AM

First Token Event

Screenshot 2023-12-07 at 11 11 00 AM

### CODE REVIEW

File: src/phoenix/core/traces.py [hunk 1]
```diff
@@ -246,7 +246,7 @@ def _consume_spans(self) -> None:
     def _process_span(self, span: pb.Span) -> None:
         span_id = SpanID(span.context.span_id)
         existing_span = self._spans.get(span_id)
-        if existing_span and existing_span.end_time:
+        if existing_span and existing_span.HasField("end_time"):
```
Comment (axiomofjoy) [inline src/phoenix/core/traces.py:249]: When an `end_time` of `None` is pass to the protobuf span, it sets it to a default timestamp that is not truthy.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -21,7 +21,7 @@ repos:
             files: \.ipynb$
             stages: [commit]
             language: system
-            entry: nbqa ruff --fix --ignore=E402 --ignore=E501
+            entry: nbqa ruff --fix --ignore=E402 --ignore=E501 --ignore=F704
```
Comment (axiomofjoy) [inline .pre-commit-config.yaml:24]: Ignore `await` outside of async function.

File: src/phoenix/core/traces.py (hunk 1 above)
Comment (RogerHYang) [reply src/phoenix/core/traces.py:249]: Good catch!

File: src/phoenix/trace/openai/instrumentor.py [hunk 3]
```diff
@@ -290,7 +297,9 @@ def __next__(self) -> ChatCompletionChunk:
                     LLM_OUTPUT_MESSAGES: _accumulate_messages(
                         chunks=self._self_chunks, num_choices=self._self_context.num_choices
                     ),  # type: ignore
-                    OUTPUT_VALUE: json.dumps([chunk.json() for chunk in self._self_chunks]),
+                    OUTPUT_VALUE: json.dumps(
+                        {"chat_completion_chunks": [chunk.dict() for chunk in self._self_chunks]}
```
Comment (RogerHYang) [inline src/phoenix/trace/openai/instrumentor.py:301]: Do we absolutely need this key? It seems to be too much "cloak-and-dagger".

**Review by RogerHYang — APPROVED** [review]

File: src/phoenix/trace/openai/instrumentor.py (hunk 3 above)
Comment (axiomofjoy) [reply src/phoenix/trace/openai/instrumentor.py]: addressed in [c5213ba](https://github.com/Arize-ai/phoenix/pull/1888/commits/c5213ba169a27b45d4903a1e78da6eb63c764719)

**Pull request merged**